src/libstore/daemon.cc

```diff
@@ -394,6 +393,34 @@ static void performOp(TunnelLogger * logger, ref<Store> store,
         break;
     }
 
+    case wopAddToStoreCANar: {
+        HashType hashAlgo;
+        std::string baseName;
+        FileIngestionMethod method;
+        {
+            uint8_t recursive;
+            std::string hashAlgoRaw;
+            from >> baseName >> recursive >> hashAlgoRaw;
+            if (recursive > (uint8_t) FileIngestionMethod::Recursive)
+                throw Error("unsupported FileIngestionMethod with value of %i; you may need to upgrade nix-daemon", recursive);
+            method = FileIngestionMethod { recursive };
+            hashAlgo = parseHashType(hashAlgoRaw);
+        }
+
+        logger->startWork();
+        std::optional<StorePath> path;
+        {
+            FramedSource narSource(from);
+            path = store->addToStoreFromDump(narSource, baseName, method, hashAlgo);
+        }
+        logger->stopWork();
+
+        if (path /* always */) {
```

Comment:
This test can be removed or replaced by `assert(path)`.

Reply:
I've changed the scope to a lambda to allow returning path without the `optional` contraption.